### PR TITLE
aioamqp: fix dependency pamqp

### PR DESCRIPTION
**Pull request by DavHau**
ZHF: #122042

aioamqp depends on pamqp < 3, but pamqp was upgraded to >3.
aioamqp is definitely not compatible with pamqp >= 3 until this is merged: https://github.com/Polyconseil/aioamqp/pull/219

Therefore I added `pamqp_2` into the package set.

Should this better be done via override?

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
I would rather apply the bump patch, and have something that works for users regardless of other packages which also may use pamqp.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122156` at f87e212d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.aioamqp</li>
    <li>python38Packages.pamqp_2</li>
    <li>python39Packages.aioamqp</li>
    <li>python39Packages.pamqp_2</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pamqp/v2.nix:23:0:
```
   |
23 |     description = "RabbitMQ Focused AMQP low-level library";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/aioamqp/default.nix:38:0:
```
   |
38 |     description = "AMQP implementation using asyncio";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/pamqp/v2.nix:22:3:
```
   |
22 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/aioamqp/default.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 { lib
 , buildPythonPackage
 , fetchFromGitHub
-, pamqp
+, pamqp_2
```
[the remark this answers is not in the document]
Comment (DavHau) [reply pkgs/development/python-modules/aioamqp/default.nix]: Oh, I wasn't aware of that. Thanks.
I now included the patch instead

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 122156 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python38Packages.aioamqp</li>    <li>python39Packages.aioamqp</li>  </ul></details>

**Pull request merged**